lib/Sema/MiscDiagnostics.cpp

```diff
@@ -221,8 +221,25 @@ static void diagSyntacticUseRestrictions(const Expr *E, const DeclContext *DC,
       }
       
       // If we have an assignment expression, scout ahead for acceptable _'s.
-      if (auto *AE = dyn_cast<AssignExpr>(E))
-        markAcceptableDiscardExprs(AE->getDest());
+      if (auto *AE = dyn_cast<AssignExpr>(E)) {
+        if (auto destExpr = AE->getDest()) {
+          markAcceptableDiscardExprs(destExpr);
+          // If the user is assigning the result of a function that returns
+          // Void to _ then warn, because that is redundant.
+          if (auto DAE = dyn_cast<DiscardAssignmentExpr>(destExpr)) {
+            if (auto CE = dyn_cast<CallExpr>(AE->getSrc())) {
+              if (CE->getCalledValue() && isa<FuncDecl>(CE->getCalledValue()) &&
+                  DAE->getType()->getRValueType()->isVoid()) {
```

Comment:
This check would be the same as `CE->getType()->isVoid()`? I wonder if it is the same, it's simpler to check for it since probably there's no need to `getRValueType`?

Reply:
Changed it now, thanks!